geometric/run_json.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python
+
+import json
+import geometric
+
+def get_qc_schema_traj(qc_schema_input, progress):
+    qc_schema_traj = []
+
+    return qc_schema_traj
+
+def parse_json_input(in_json):
+    """
+    Parse an input json file into options, example:
+    {
+        "schema_name": "qc_schema_optimization_input",
+        "schema_version", 1,
+        "geometric_options": {
+            "coordsys": "tric",
+            "conv": 1.e-7
+        }
+        "input_specification": qc_schema_input,
+    }
+
+    qc_schema_input = {
+        "schema_version": "v0.1",
+        "molecule": {
+            "geometry": [
+                0.0, 0.0, -0.1294769411935893, 0.0,
+               -1.494187339479985, 1.0274465079245698,
+                0.0, 1.494187339479985, 1.0274465079245698
+            ],
+            "symbols": ["O", "H", "H"],
+            "connectivity": [[0, 1, 1], [0, 2, 1]]
+        },
+        "driver": "gradient",
+        "model": {
+            "method": "UFF",
+            "basis": None
+        },
+        "keywords": {},
+        "program": "rdkit"
+    }
+    """
+    d = json.load(open(in_json))
+    input_opts = d['geometric_options']
+    # insert `fix_orientation` and `fix_com`
+    d['input_specification'].update({
+        'fix_orientation': True,
+        'fix_com': True
+    })
+    # Here we force the use of qcengine because other engines don't support qc schema
+    input_opts.update({
+        'qcengine': True,
+        'qcschema': d['input_specification']
+    })
+    return input_opts
+
+def write_json_output(in_json, schema_traj, out_json):
+    # copy the input json data
+    out_json_data = json.load(open(in_json))
+    out_json_data.update({
+        "trajectory": schema_traj,
+        "energies": [s['properties']['return_energy'] for s in schema_traj],
+        "final_molecule": schema_traj[-1]
+    })
+    json.dump(out_json_data, open(out_json, 'w'), indent=2)
+
+def geometric_run_json(in_json, out_json):
```

Comment:
This is the main function for the json interface. It is a simple version of the run_optimize() function in optimize.py.